plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/AwsS3Service.java

```diff
@@ -153,5 +153,6 @@
         Setting<String> ENDPOINT_SETTING = Setting.simpleString("cloud.aws.s3.endpoint", Property.NodeScope);
     }
 
-    AmazonS3 client(String endpoint, Protocol protocol, String region, String account, String key, Integer maxRetries);
+    AmazonS3 client(String endpoint, Protocol protocol, String region, String account, String key, Integer maxRetries,
+                    Boolean pathStyleAccess);
```

Comment:
In another PR, you asked me the opposite: rely on AWS SDK defaults.

Here the Javadoc for this feature says:

> The default behaviour is to detect which access style to use based on
> the configured endpoint (an IP will result in path-style access) and
> the bucket being accessed (some buckets are not valid DNS names).
> Setting this flag will result in path-style access being used for all requests.

So my guess is that we should let the SDK decide. Do you agree?